Add documentation for `group_by_use_nulls` setting (#46460)
ClickHouse/ClickHouse: docs/en/operations/settings/settings.md

Default value: 0.

## group_by_use_nulls {#group_by_use_nulls}

> Changes the way [GROUP BY clause](../../sql-reference/statements/select/group-by.md) treats the types of aggregation keys.

Reviewer: ```suggestion
Changes the way the [GROUP BY clause](../../sql-reference/statements/select/group-by.md) treats the types of aggregation keys.
```

> When `ROLLUP`, `CUBE` or `GROUPING SETS` are used, some aggregation keys may not be used to produce some result raws.

Reviewer: ```suggestion
When the `ROLLUP`, `CUBE`, or `GROUPING SETS` clauses are used, some aggregation keys may not be used to produce some result rows.
```

> Columns for these keys are filled with either default value or `NULL` in corresponding raws depending on this setting.

Reviewer: ```suggestion
Columns for these keys are filled with either default value or `NULL` in corresponding rows depending on this setting.
```

Possible values:

-   0 — The default value for the aggregation key type is used to produce missing values.

> -   1 — ClickHouse executes `GROUP BY` the same way as the SQL standard says. The types of aggregation keys are converted to [Nullable](../../sql-reference/data-types/nullable.md/#data_type-nullable). Columns for corresponding aggregation keys are filled with [NULL](../../sql-reference/syntax.md) for raws that didn't use it.

Reviewer: ```suggestion
-   1 — ClickHouse executes `GROUP BY` the same way as the SQL standard says. The types of aggregation keys are converted to [Nullable](/docs/en/sql-reference/data-types/nullable.md/#data_type-nullable). Columns for corresponding aggregation keys are filled with [NULL](/docs/en/sql-reference/syntax.md) for rows that didn't use it.
```

> -   1 — ClickHouse executes `GROUP BY` the same way as the SQL standard says. The types of aggregation keys are converted to [Nullable](/docs/en/sql-reference/data-types/nullable.md/#data_type-nullable). Columns for corresponding aggregation keys are filled with [NULL](/docs/en/sql-reference/syntax.md) for rows that didn't use it.

Reviewer 2: ```suggestion
-   1 — ClickHouse executes `GROUP BY` the same way as the SQL standard says. The types of aggregation keys are converted to [Nullable](/docs/en/sql-reference/data-types/nullable.md). Columns for corresponding aggregation keys are filled with [NULL](/docs/en/sql-reference/syntax.md#null) for rows that didn't use it.
```

Default value: 0.

See also:

> -   [GROUP BY clause](../../sql-reference/statements/select/group-by.md)

Reviewer: ```suggestion
-   [GROUP BY clause](/docs/en/sql-reference/statements/select/group-by.md)
```
